tests/unit/test_elgamal.py

```diff
@@ -0,0 +1,63 @@
+from electionguard import ElGamalKeyPair
+from electionguard.elgamal import (
+    ElGamalPublicKey,
+    ElGamalSecretKey,
+    hashed_elgamal_encrypt,
+)
+from electionguard.group import ElementModQ
+from electionguard.serialize import PaddedDataSize, _add_padding, _remove_padding
+from electionguard.utils import get_optional
+from tests.base_test_case import BaseTestCase
+
+
+class TestElgamal(BaseTestCase):
+    """Test decryption methods"""
+
+    def test_hashed_elgamal_with_session_key_that_starts_with_0(self) -> None:
+        kp = ElGamalKeyPair(
+            secret_key=ElGamalSecretKey("02"),
+            public_key=ElGamalPublicKey("A147CA31DE0F48C1"),
+        )
+        plaintext = b"\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00"
+
+        # these values produce a session_key 2B00, which produces encrypted
+        # data that begins with a 0 byte. When hashed_elgamal_encrypt() returns that
+        # data it calls bytes_to_hex() which truncates leading zero's. That produces
+        # 255 bytes instead of 256 bytes and decrypt() much pad a byte to account for it.
+        hmac_nonce = ElementModQ("D8E1")
+        hmac_seed = ElementModQ("02F1")
+
+        self.do_hashed_elgamal(kp, plaintext, hmac_nonce, hmac_seed)
+
+    def test_hashed_elgamal_with_session_key_that_starts_with_1(self) -> None:
+        kp = ElGamalKeyPair(
+            secret_key=ElGamalSecretKey("02"),
+            public_key=ElGamalPublicKey("A147CA31DE0F48C1"),
+        )
+        plaintext = b"\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00"
+
+        # these values produce a session_key EE19 which produces encrypted data that
+        # begins with a 1 byte and bytes_to_hex() does not truncate anything
+        hmac_nonce = ElementModQ("AB29")
+        hmac_seed = ElementModQ("2179")
+
+        self.do_hashed_elgamal(kp, plaintext, hmac_nonce, hmac_seed)
+
+    def do_hashed_elgamal(
+        self,
+        kp: ElGamalKeyPair,
+        plaintext: bytes,
+        hmac_nonce: ElementModQ,
+        hmac_seed: ElementModQ,
+    ) -> None:
+        padded_plaintext = _add_padding(plaintext, PaddedDataSize.Bytes_512)
```

Comment:
This is a good example of what I was discussing before with the use of PaddedDataSize here. 

1. Pad `plaintext` to `512` bytes
2. `hashed_elgamal_decrypt` while adding one more bit of padding to make sure it's `512`

Reply:
We're only working in 512 with this whole encryption so we should have it made much more obvious the whole way through.

Reply:
https://github.com/microsoft/electionguard-python/blob/5b73f75cc54f63837bf7cd6328adff208e6dca62/src/electionguard/serialize.py#L51-L55

perhaps this
```py
 class DataSize(IntEnum): 
     """Define the sizes for data.""" 
  
     Bytes_512 = 512
     
def get_padded_data_size(data_size: DataSize): 
      return DataSize - PAD_INDICATOR_SIZE
```

Reply:
This also works but it'll expose the items for modification while it's running.

```py
 DATA_SIZE = 512
 PAD_INDICATOR_SIZE = 2
 PADDED_DATA_SIZE = DATA_SIZE - PAD_INDICATOR_SIZE
 ```

Reply:
This line is probably relevant in making decisions
https://github.com/microsoft/electionguard-python/blob/5b73f75cc54f63837bf7cd6328adff208e6dca62/src/electionguard/serialize.py#L91